android/src/main/kotlin/com/icapps/background_location_tracker/service/LocationUpdatesService.kt

```diff
@@ -100,17 +101,26 @@ internal class LocationUpdatesService : Service() {
         // and binds with this service. The service should cease to be a foreground service
         // when that happens.
         Logger.debug(TAG, "OnBind")
-        stopForeground(true)
+        stopForeground()
         changingConfiguration = false
         return binder
     }
 
+    @Suppress("DEPRECATION")
+    private fun stopForeground() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            stopForeground(STOP_FOREGROUND_REMOVE)
+        } else {
+            stopForeground(true)
+        }
+    }
+
     override fun onRebind(intent: Intent) {
         // Called when a client (MainActivity in case of this sample) returns to the foreground
         // and binds once again with this service. The service should cease to be a foreground
         // service when that happens.
         Logger.debug(TAG, "OnRebind")
-        stopForeground(true)
+        stopForeground()
```

Comment:
Perhaps we should try to avoid naming our methods the same as the framework methods